### ISSUE TITLE
While navigating a directory listing, nested directories don't respond correctly.

### ISSUE THREAD

**Issue by esmevane** [esmevane/duckling#6]
When I start duckling up, using its own home directory as the root public directory, my first visit looks like this:

<img width="233" alt="screen shot 2016-11-15 at 4 19 33 pm" src="https://cloud.githubusercontent.com/assets/263460/20326694/74221eae-ab4f-11e6-8dfe-1800be799bee.png">

Looking one level deeper, following `src/`, I see this:

<img width="273" alt="screen shot 2016-11-15 at 4 21 40 pm" src="https://cloud.githubusercontent.com/assets/263460/20326736/9b1ff5b2-ab4f-11e6-8af5-9afe55e2e713.png">

In this page, `main` and `test` are directories as well, and they should have a trailing slash (at the very least).  But also, when I click on one (say, `test`):

<img width="255" alt="screen shot 2016-11-15 at 4 23 12 pm" src="https://cloud.githubusercontent.com/assets/263460/20326786/d46b0f0a-ab4f-11e6-94b4-24f10b6d1f5b.png">

In other words, it's taking me to the topmost level - there doesn't seem to be any recognition of nested directories at all.
